test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/Services/NuGetProjectManagerServiceTests.cs

```diff
@@ -535,6 +556,827 @@ public async Task GetUpdateActionsAsync_WithPackageReferenceProject_WhenUpdating
+
+                var packageSpecs = await testProject.GetPackageSpecsAsync(context);
+
+                // Package directories
+                var sources = new List<PackageSource>();
+                var packagesDir = new DirectoryInfo(Path.Combine(testDirectory, "globalPackages"));
+                var packageSource = new DirectoryInfo(Path.Combine(testDirectory, "packageSource"));
+                packagesDir.Create();
+                packageSource.Create();
+                sources.Add(new PackageSource(packageSource.FullName));
+
+                Initialize(sources);
+
+                _solutionManager.NuGetProjects.Add(testProject);
+
+                var logger = new TestLogger();
+                var request = new TestRestoreRequest(packageSpecs[0], sources, packagesDir.FullName, logger)
+                {
+                    LockFilePath = Path.Combine(testDirectory, "obj", "project.assets.json")
+                };
+
+                await SimpleTestPackageUtility.CreateFullPackageAsync(packageSource.FullName, "packageC", "2.1.43");
+                await SimpleTestPackageUtility.CreateFullPackageAsync(
+                    packageSource.FullName,
+                    "packageB",
+                    "1.0.0",
+                    new PackageDependency[]
+                    {
+                        new PackageDependency("packageC", VersionRange.Parse("2.1.43"))
+                    });
+                await SimpleTestPackageUtility.CreateFullPackageAsync(
+                    packageSource.FullName,
+                    "packageA",
+                    "2.15.3",
+                    new PackageDependency[]
+                    {
+                        new PackageDependency("packageB", VersionRange.Parse("1.0.0"))
+                    });
+
+                var command = new RestoreCommand(request);
+                RestoreResult result = await command.ExecuteAsync();
+                await result.CommitAsync(logger, CancellationToken.None);
+                Assert.True(result.Success);
+
+                // Act
+                var installedAndTransitive = await _projectManager.GetInstalledAndTransitivePackagesAsync(new[] { projectId }, CancellationToken.None);
+
+                // packageB and packageC are transitive dependencies
+                installedAndTransitive.TransitivePackages.Should().HaveCount(2);
+
+                var transitivePackageB = installedAndTransitive.TransitivePackages.Where(pkg => pkg.Identity.Id == "packageB").First();
+                transitivePackageB.TransitiveOrigins.Should().HaveCount(1);
+                var transitiveOriginB = transitivePackageB.TransitiveOrigins.First();
+                Assert.Equal("packageA", transitiveOriginB.Identity.Id);
+                Assert.Equal(new NuGetVersion("2.15.3"), transitiveOriginB.Identity.Version);
+
+                var transitivePackageC = installedAndTransitive.TransitivePackages.Where(pkg => pkg.Identity.Id == "packageC").First();
```

Comment:
```suggestion
                var transitivePackageC = installedAndTransitive.TransitivePackages.Where(pkg => pkg.Identity.Id == "packageC").Single();
```

...to also test that there's only 1 occurrence of `packageC` as a transitive?

Reply:
Please note Single has perf cost over First here.